app/(server)/_features/event/repository.ts

```diff
@@ -38,13 +38,55 @@ export class EventRepo implements iEventRepo {
     else return undefined;
   }
 
+  /**
+   * Retrive published events that not yet started and drafted and cancelled events by the user.
+   */
+  async getMyEvents(userID: number, page: number, limit: number) {
+    page = page - 1;
+
+    if (page < 0) {
+      page = 0;
+    }
+
+    if (limit <= 0) {
+      limit = 10;
+    }
+
+    const filter = sql`
+    ${events.createdBy} = ${userID} AND(
+    (${events.status} = ${'published'} AND ${events.endTime} >= NOW()) 
+    OR ${events.status} = ${'cancelled'} 
+    OR ${events.status} = ${'draft'})`;
+
+    const res = await db.query.events.findMany({
+      where: filter,
+      limit: limit,
+      offset: page * limit,
+    });
```

Comment:
@GagahGK , I think we need to ensure the deleted events are not included in the response. I have deleted some events, but when I tried to fetch to the endpoint, I still got the deleted events data.

Reply:
@GagahGK Btw, when event's status is `draft` and the `deletedAt` field is already filled, the deleted events are still included in the response. I think because the filter only check if the event status is draft, and not checking whether the event has been deleted, that's why the deleted events are still included in the response to client

Reply:
thankyou, i forgot to filter out the soft deleted event